### PR TITLE
update to 2.9.0.post0

### PR DESCRIPTION
**Pull request by cbouss**
Re-update, because the previous version was fixed in the url.
Remove older patches and comments.

### CODE REVIEW

File: recipe/new_tzfile.patch
Comment (JeanChristopheMorinPerso) [inline recipe/new_tzfile.patch:1]: Umm, I think SF still want this file. Obiviously, this is only for their version and not the version on defaults.

**Review by JeanChristopheMorinPerso — CHANGES_REQUESTED** [review]

File: recipe/new_tzfile.patch
Comment (cbouss) [reply recipe/new_tzfile.patch:1]: I opened a ticket to track this. https://anaconda.atlassian.net/browse/PKG-4850
This PR is for our main channel though.

**Review by JeanChristopheMorinPerso — APPROVED** [review]
Approving, but the SF version will have to be sorted out quickly.

**Review by Jrice1317 — APPROVED** [review]

**Comment by kynan** [comment]
@cbouss @JeanChristopheMorinPerso Could you please confirm why this is not a `noarch` build?

**Comment by JeanChristopheMorinPerso** [comment]
@kynan We always build our packages this way. We partly do this because of how we build and test packages at Anaconda.

**Comment by danpetry** [comment]
It was a policy decision made a while back, because there were too many edge cases with `python: noarch`, i.e. python packages not really being version-independent, which needed to be handled. We took the hit of having to build packages for each python version over the need to have all the edge cases, with specific version selectors, handled manually in the feedstocks.

**Comment by danpetry** [comment]
Also, then they get built and tested explicitly on all python interpreter versions.

**Comment by kynan** [comment]
Thanks for confirming. That rationale makes sense :)

**Comment by sam-s** [comment]
I still observe the same problem after `conda install --force-reinstall python-dateutil` - files stamped with `2024-05-07`, `conda list python-dateutil` reports `2.9.0post0` while  `_version.py` has `2.8.2`

**Comment by danpetry** [comment]
could you give a bit more info (eg output of `conda install --force-reinstall python-dateutil`, full build string of your python-dateutil package, whether you're using defaults or conda-forge?

**Comment by sam-s** [comment]
> could you give a bit more info (eg output of `conda install --force-reinstall python-dateutil`, full build string of your python-dateutil package, whether you're using defaults or conda-forge?

How do I know _"whether I'm using defaults or conda-forge"_?

```
$ conda install --force-reinstall python-dateutil
Channels:
 - https://artprod.dev.bloomberg.com/artifactory/api/conda/bb-conda
 - https://artprod.dev.bloomberg.com/artifactory/api/conda/conda-forge
 - https://artprod.dev.bloomberg.com/artifactory/bb-bqnt/conda-repo
 - https://artprod.dev.bloomberg.com/artifactory/bqnt-conda-dev
 - conda-forge
 - defaults
Platform: win-64
Collecting package metadata (repodata.json): done
Solving environment: done

## Package Plan ##

  environment location: $HOME\AppData\Local\miniconda3\envs\c312

  added / updated specs:
    - python-dateutil



Proceed ([y]/n)?


Downloading and Extracting Packages:

Preparing transaction: done
Verifying transaction: done
Executing transaction: done
$ conda list --show-channel-urls dateutil
# packages in environment at $HOME\AppData\Local\miniconda3\envs\c312:
#
# Name                    Version                   Build  Channel
python-dateutil           2.9.0post0      py312haa95532_0    defaults
```

**Comment by danpetry** [comment]
Thanks. can you try `conda install --force-reinstall python-dateutil --override-channels -c defaults`?

**Comment by sam-s** [comment]
Thank you, this command did result in
```
$ conda list --show-channel-urls dateutil
# packages in environment at $HOME\AppData\Local\miniconda3\envs\c312:
#
# Name                    Version                   Build  Channel
python-dateutil           2.9.0post0      py312haa95532_2    defaults
```
however, `python-dateutil` was already installed from `defaults` -- why do I need to `--override-channels`?!

**Comment by danpetry** [comment]
You've also got other channels configured there. If you were to uninstall then reinstall, it would pick up one of your higher priority channels instead. I think the channel priority is interfering with `--force-reinstall`.

**Comment by danpetry** [comment]
```
Channels:
 - https://artprod.dev.bloomberg.com/artifactory/api/conda/bb-conda
 - https://artprod.dev.bloomberg.com/artifactory/api/conda/conda-forge
 - https://artprod.dev.bloomberg.com/artifactory/bb-bqnt/conda-repo
 - https://artprod.dev.bloomberg.com/artifactory/bqnt-conda-dev
 - conda-forge
 - defaults
```

**Comment by sam-s** [comment]
Both before and after `python-dateutil` was installed from `defaults` - how could the other channels affect the choice?

**Comment by danpetry** [comment]
Hi @sam-s, apologies for the delay here.
This could be a bug in conda. I've opened a bug report [here](https://github.com/conda/conda/issues/14372). Do you have any further questions I can help with?

**Comment by danpetry** [comment]
The behaviour of `--force-reinstall` should be the same as if you were to uninstall and then reinstall python-dateutils. This should result in a package from conda-forge instead of defaults, because you have conda-forge configured as a higher-priority channel.

**Pull request merged**